src/util/safelywritablefile.h

```diff
@@ -31,6 +31,8 @@ class SafelyWritableFile final {
 
     bool commit();
 
+    bool useTempFileWithPrefix(const QString& coverArtSuffix);
```

Comment:
```suggestion
    bool useTempFileWithPrefix();
    bool useTempFileWithSuffix();
```

This functions should be unrelated to coverArts. This functions should instruct the SafelyWritableFile class to reserve a a new temporary file name without copy the original file. Since the Prefix is added to the original file name which already has the suffix, we do not need the parameter. 

`useTempFileWithSuffix();` can be added for symmetry and can be used internally from the constructor.  

Idea: Since this functions must not called after  writing, we may consider to add another boolean flag or enum to control that from the constructor. (for the next step)

Reply:
I think we should pass the original File name in the constructor and fix the bug.

Reply:
You just need to reorder the constructor that your call happens after:
https://github.com/mixxxdj/mixxx/blob/36e86965debbdd5a0a78024d2efae54f8c8165e4/src/util/safelywritablefile.cpp#L89